mptt/admin.py

```diff
@@ -172,6 +171,16 @@ def changelist_view(self, request, *args, **kwargs):
 
         return response
 
+    def get_data_before_update(self, request, cut_item, pasted_on):
+        mptt_opts = self.model._mptt_meta
+        mptt_attr_fields = ("parent_attr", "left_attr", "right_attr", "tree_id_attr", "level_attr")
+        mptt_fields = [getattr(mptt_opts, attr) for attr in mptt_attr_fields]
+        return {k: getattr(cut_item, k) for k in mptt_fields}
+
+    def get_mode_node_change_message_log(self, request, cut_item, pasted_on, data_before_update):
```

Comment:
Shouldn't this be `get_move_node_change_message`? move, not mode; also, `_log` seems unnecessary.

Looks good apart from that!

Reply:
I missed that typo. thanks!